e2e/tests/validator.go

```diff
@@ -18,7 +18,19 @@ var (
 )
 
 func init() {
-	rpcURL = fmt.Sprintf("http://%s:%s", runtime.GetEnv("RPC_HOST", "pocket-validators"), defaults.DefaultRPCPort)
+	// set the rpcURL based on the environment while still supporting overriding of the RPC_HOST.
+	var rpcHost string
+	// TECHDEBT: if we intend to support running the e2e tests in both tilt/k8s and
+	// docker compose, then we need to understand why the `KUBERNETES_SERVICE_HOST`
+	// isn't set even when running in k8s. See: `test_go('e2e-tests', ...` in the
+	// Tiltfile.
+	// if runtime.IsProcessRunningInsideKubernetes() {
```

Comment:
Is this a WIP to be checked with @okdas in this PR?

Reply:
If we don't care about running e2e tests in docker compose, I would just clean this up.